lib/workflows/dc-api-workflow.js

```diff
@@ -0,0 +1,331 @@
+  }
+
+  async authorizationRequest(req, res) {
+    const rp = req.rp;
+    const exchange = await this.getExchange({rp, id: req.params.exchangeId});
+
+    logUtils.presentationStart(rp?.clientId, exchange?.id);
+    if(!exchange || exchange?.workflowId !== req.params.workflowId) {
+      const errorMessage = 'Exchange not found';
+      logUtils.presentationError(rp?.clientId, 'unknown', errorMessage);
+      res.status(404).send({message: errorMessage});
+      return;
+    }
+    if(exchange.state !== 'pending' && exchange.state !== 'active') {
+      const errorMessage = `Exchange in state ${exchange.state}`;
+      logUtils.presentationError(rp?.clientId, exchange.id, errorMessage);
+      res.status(400).send(errorMessage);
+      return;
+    }
+    try {
+      const sessionId =
+        exchange.variables.dcApiSession.session_creation_response.id;
+      const sessionSecret =
+        exchange.variables.dcApiSession.session_creation_response.client_secret;
+
+      if(!sessionSecret || !sessionId) {
+        res.status(500).send({message: 'Session data missing from exchange'});
+        return;
+      }
+
+      const requests = await this.dcApi.initiate_request(
+        sessionId,
+        sessionSecret,
+        JSON.parse(rp.workflow.dcApiRequest),
+        req.headers['user-agent'],
+      );
+
+      await database.collections.Exchanges.updateOne(
+        {id: exchange.id},
+        {
+          $set: {state: 'active'},
+        },
+      );
+
+      res.send(requests);
+    } catch(error) {
+      logUtils.presentationError(rp?.clientId, exchange.id, error.message);
+      logger.error(error.message, {error});
+      res.sendStatus(500);
+    }
+    return;
+  }
+
+  async authorizationResponse(req, res) {
+    try {
+      const rp = req.rp;
+
+      const exchange = await this.getExchange({
+        rp,
+        id: req.params.exchangeId,
+      });
+
+      if(!exchange || exchange?.workflowId !== req.params.workflowId) {
+        const errorMessage = 'Exchange not found';
+        logUtils.presentationError(rp?.clientId, 'unknown', errorMessage);
+        res.status(404).send({message: errorMessage});
+        return;
+      }
+
+      if(exchange.state !== 'active') {
+        const errorMessage = `Exchange in state ${exchange.state}`;
+        logUtils.presentationError(rp?.clientId, exchange.id, errorMessage);
+        res.status(400).send(errorMessage);
+        return;
+      }
+
+      const results = await this.dcApi.submit_response(
+        exchange.variables.dcApiSession.session_creation_response.id,
+        exchange.variables.dcApiSession.session_creation_response.client_secret,
+        req.body,
+      );
+
+      // Generate OIDC authorization code for Keycloak/OAuth flows
+      const oidcCode = await createId();
+
+      await database.collections.Exchanges.updateOne(
+        {id: exchange.id},
+        {
+          $set: {
+            state: 'complete',
+            step: 'default',
```

Comment:
Shouldn't need to update step, look earlier at where step may have been previously set